File: app/src/main/java/com/karthik/myweather/data/entities/CityWeather.java
package com.karthik.myweather.data.entities;


import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

public class CityWeather {
    @Embedded
    public City city;

    @Relation(
            parentColumn = "locationId",
            entityColumn = "locationId")
    public List<Weather> weatherList;
}

File: app/src/main/java/com/karthik/myweather/data/entities/CityWeather.kt
package com.karthik.myweather.data.entities

import androidx.room.Embedded
import androidx.room.Relation

data class CityWeather(
    @Embedded
    var city: City? = null,
    @Relation(parentColumn = "locationId", entityColumn = "locationId")
    var weatherList: List<Weather>? = null)
